collections/_posts/2022-10-26-GeneralizationInFederatedLearning.md

```diff
@@ -0,0 +1,102 @@
+
+**They further define the *empirical risk* on the participating training client data with**
+
+$$ F_{part\_train}(w) := \frac{1}{|\hat{C}|}\underset{c\in \hat{C}}{\sum}[\rho_c*(\frac{1}{|\hat{\Xi}_c|}\underset{\xi\in\hat{\Xi}_c}{\sum} f(w,\xi))]$$
+
+**And the *semi-empirical risk* on the participating validation client data with**
+
+$$ F_{part\_val}(w) := \frac{1}{|\hat{C}|}\underset{c\in\hat{C}}{\sum}[\rho_c*(\mathbb{E}_{\xi\sim D_c}f(w,\xi))]$$
+
+They finally define two levels of generalization gap:
+* **The out-of-sample gap: $$ F_{part\_val}(w) - F_{part\_train}(w)$$**
+* **The partipation gap: $$ F_{unpart}(w) - F_{part\_val}(w)$$**
+
+## Highlighted properties
+* *Participation gap is an intrinsic property of FL due to heterogeneity*: the gap will not exist if data is homogeneous,
+* *Participation gap can quantify client diversity* (comparable between tasks),
+* *Participation gap can measure overfitting on the population distribution*,
+* *Participation gap can quantify model robustness to unseen clients*,
+* *Participation gap can quantify the incentive for clients to participate*.
+
+# The three-way validation split
+![](/collections/images/generalization_federated_learning/three_split.jpg)
+
+Each column corresponds to the dataset of one client. A
+dataset is split into participating training, participating
+validation, and unparticipating data, which enables
+separate measurement of out-of-sample and participation gaps.
+
+# Two levels of generalization gap in practice
+
+* With federated learning
+
+![](/collections/images/generalization_federated_learning/generalization_gaps_fl.jpg)
+
+* But even with centralized training
+
+![](/collections/images/generalization_federated_learning/centralized_participation_gap.jpg)
+
+# Label partitioning is not enough to simulate heterogeneity
+
+* To explore federated learning when there are no large federated datasets accessible, one can synthetically partition a public dataset.
+* The authors show that simulating heterogeneity of a federated dataset by generating partitions with different label distributions alter the training processes compared to a naturally heterogeneous federated dataset.
+
+![](/collections/images/generalization_federated_learning/learning_progress_natural_label_part.jpg)
+
+# Semantic partitioning
+
+The proposed partitioning protocol is the following for a classification task:
+* For each label, extract semantic features with a pretrained neural network. Fit a Gaussian Mixture Model to cluster pretrained embeddings into $$C$$ groups: **intra-client intra-label consistency**.
+* Then, the aim is to compute an optimal multi-partite matching with cost-matrix defined by KL-divergence between the Gaussian clusters. They approximate this optimal by progressively solving the optimal bipartite matching at each time for randomly-chosen label pairs: **intra-client inter-label consistency**.
```

Comment:
From a quick search, _multi-partite_ seems to rather be written as one word: _multipartite_. Unless you're using a specific definition of the word that is normally written this way in the literature?
```suggestion
* Then, the aim is to compute an optimal multipartite matching with cost-matrix defined by KL-divergence between the Gaussian clusters. They approximate this optimal by progressively solving the optimal bipartite matching at each time for randomly-chosen label pairs: **intra-client inter-label consistency**.
```